tests/main/security-device-cgroups-serial-port/task.yaml

```diff
@@ -0,0 +1,55 @@
+summary: Ensure that the device cgroup works properly for serial-port.
+
+# We don't run the native kernel on these distributions yet so we can't
+# load kernel modules coming from distribution packages yet.
+systems: [-fedora-*, -opensuse-*, -debian-unstable-*]
+
+prepare: |
+    # create serial devices if they don't exist
+    if [ ! -e /dev/ttyS4 ]; then
+        mknod /dev/ttyS4 c 4 68
+        touch /dev/ttyS4.spread
+    fi
+
+restore: |
+    if [ -e /dev/ttyS4.spread ]; then
+        rm -f /dev/ttyS4 /dev/ttyS4.spread
+    fi
+
+    udevadm control --reload-rules
+    udevadm trigger
+
+execute: |
+    echo "Given a snap is installed"
+    . $TESTSLIB/snaps.sh
+    install_local test-snapd-tools
+
+    echo "Then the device is not assigned to that snap"
+    ! udevadm info /dev/ttyS4 | MATCH "E: TAGS=.*snap_test-snapd-tools_env"
+
+    echo "And the device is not shown in the snap device list"
+    # FIXME: this is, apparently, a layered can of worms. Zyga says he needs to fix it.
```

Comment:
Holly molly, this is a copy-pasted can of worms. I need to look here, I presume.

Reply:
I did copy/paste the comment because I didn't want you to come back to the other only to forget this one. :) Honestly, I'm not sure why we should spend time fixing it. It isn't the prettiest thing in the world, but it also isn't that bad. We update the rules file, run udevadm, and make sure the device is added to the snap-specific device cgroup.